tests/lib.rs

```diff
@@ -352,3 +352,61 @@ fn test_list_flag_rendering() {
         .success()
         .stdout("bar\nbaz\nfoo\nqux\n");
 }
+
+#[test]
+fn test_autoupdate_cache() {
+    let testenv = TestEnv::new();
+
+    // The first time, if automatic updates are disabled, the cache should not be found
+    testenv
+        .command()
+        .args(&["--list"])
+        .assert()
+        .failure()
+        .stderr(contains("Cache not found. Please run `tldr --update`."));
+
+    let config_file_path = testenv.config_dir.path().join("config.toml");
+    let cache_file_path = testenv.cache_dir.path().join("tldr-master");
+
+    // Activate automatic updates, set the auto-update interval to 24 hours
+    let mut config_file = File::create(&config_file_path).unwrap();
+    config_file
+        .write("[updates]\nauto_update = true\nauto_update_interval_hours = 24".as_bytes())
+        .unwrap();
+    config_file.flush().unwrap();
+
+    // Helper function that runs `tldr --list` and asserts that the cache is automatically updated
+    // or not, depending on the value of `expected`.
+    let check_cache_updated = |expected| {
+        let assert = testenv.command().args(&["--list"]).assert().success();
+        let pred = contains("Successfully updated cache");
+        if expected {
+            assert.stdout(pred)
+        } else {
+            assert.stdout(pred.not())
+        };
+    };
+
+    // The cache is updated the first time we run `tldr --list`
+    check_cache_updated(true);
+
+    // The cache is not updated with a subsequent call
+    check_cache_updated(false);
+
+    let (_, orig_mtime) = utime::get_file_times(&cache_file_path).unwrap();
+
+    // We update the modification and access times such that they are about 23 hours from now.
+    // auto-update interval is 24 hours, the cache should not be updated
+    let new_mtime = orig_mtime - 82_800;
+    utime::set_file_times(&cache_file_path, new_mtime, new_mtime).unwrap();
```

Comment:
Since the `utime` library expects a `u64`, I had to set the mtime relative to `orig_mtime`, as opposed to something more explicit like `let new_mtime = std::time::SystemTime::now() - Duration::from_secs(82_800)`.

The [`filetime`](https://crates.io/crates/filetime) crate has a [`FileTime::from_system_time()`](https://docs.rs/filetime/0.2.9/filetime/struct.FileTime.html#method.from_system_time) function and could be used to set the atime and mtime. The crate is actively maintained, unlike `utime` (last updated in October 2018). This is not critical since `utime` works fine here.